--- less-392/screen.c.Foption	2005-09-17 19:17:02.000000000 +0200
+++ less-392/screen.c	2005-10-21 16:25:16.000000000 +0200
@@ -233,6 +233,7 @@ extern int wscroll;
 extern int screen_trashed;
 extern int tty;
 extern int top_scroll;
+extern int quit_if_one_screen;
 #if HILITE_SEARCH
 extern int hilite_search;
 #endif
@@ -1547,7 +1548,7 @@ deinit()
 #if !MSDOS_COMPILER
 	if (!no_keypad)
 		tputs(sc_e_keypad, sc_height, putchr);
-	if (!no_init)
+	if (!no_init && !quit_if_one_screen)
 		tputs(sc_deinit, sc_height, putchr);
 #else
 	/* Restore system colors. */
